static/skywire-manager-src/src/app/services/vpn-client.service.ts

```diff
@@ -0,0 +1,787 @@
+      return CheckPkResults.SamePkStopped;
+    }
+
+    return CheckPkResults.Ok;
+  }
+
+  /**
+   * Changes the currently selected server to the one set in this.requestedServer. After that,
+   * the VPN is started.
+   */
+  private processServerChange() {
+    if (this.dataSubscription) {
+      this.dataSubscription.unsubscribe();
+    }
+
+    const data = { pk: this.requestedServer.pk };
+    if (this.requestedPassword) {
+      data['passcode'] = this.requestedPassword;
+    } else {
+      data['passcode'] = '';
+    }
+
+    // Mark the service as busy, stop updating the VPN state and inform about the changes.
+    this.stopContinuallyUpdatingData();
+    this.working = true;
+    this.sendUpdate();
+
+    // Make the changes in the local visor.
+    this.dataSubscription = this.appsService.changeAppSettings(
+      this.nodeKey,
+      this.vpnClientAppName,
+      data,
+    ).subscribe(
+      () => {
+        // Save the changes locally.
+        this.vpnSavedDataService.modifyCurrentServer(this.requestedServer);
+
+        // Make the service work normally again.
+        this.requestedServer = null;
+        this.requestedPassword = null;
+        this.working = false;
+
+        // Start the VPN.
+        this.start();
+      }, (err: OperationError) => {
+        // Inform about the error.
+        err = processServiceError(err);
+        this.snackbarService.showError('vpn.server-change.backend-error', null, false, err.originalServerErrorMsg);
+
+        // Make the service work normally again.
+        this.working = false;
+        this.requestedServer = null;
+        this.requestedPassword = null;
+        this.sendUpdate();
+        this.updateData();
+      }
+    );
+  }
+
+  /**
+   * Checks and configures the local visor to make it posible to start or stop the VPN and then
+   * starts or stops it.
+   * @param startApp If the VPN must be started or stopped.
+   */
+  private checkBeforeChangingAppState(startApp: boolean) {
+    // Cancel if the service is busy.
+    if (this.working) {
+      return;
+    }
+
+    // Mark the service as busy, stop updating the VPN state and inform about the changes.
+    this.stopContinuallyUpdatingData();
+    this.working = true;
+    this.sendUpdate();
+
+    // If the VPN is going to be stopped, just continue with the process, as no config is needed.
+    if (!startApp) {
+      this.changeAppState(startApp);
+
+      return;
+    }
+
+    if (this.dataSubscription) {
+      this.dataSubscription.unsubscribe();
+    }
+
+    // Get the current general state of the local visor.
+    this.dataSubscription = this.apiService.get(`visors/${this.nodeKey}`).pipe(
+      mergeMap(nodeInfo => {
+        // Check if the local visor already has a transport for connecting with the server.
+        let transportFound = false;
+        if (nodeInfo.transports && nodeInfo.transports.length > 0) {
+          (nodeInfo.transports as any[]).forEach(transport => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/skycoin/skywire/pullRequest?prid=7163595)
```suggestion
          (nodeInfo.transports as any[]).forEach((transport) => {
```